src/test/java/io/jenkins/plugins/util/AgentFileVisitorTest.java

```diff
@@ -118,6 +118,21 @@ void shouldReturnMultipleResults() {
         assertThat(actualResult.getLog().getErrorMessages()).containsExactly("Errors during parsing",
                 "Skipping file 'empty.txt' because Jenkins has no permission to read the file",
                 "Skipping file 'not-readable.txt' because it's empty");
+
+        StringScanner scannerTwo = new StringScanner(PATTERN, "UTF-8", true, false,
+                fileSystemFacade);
+
+        ScannerResult<String> actualResultTwo = scannerTwo.invoke(workspace, null);
+        assertThat(actualResultTwo.getResults()).containsExactly(CONTENT + 1, CONTENT + 2);
+        assertThat(actualResultTwo.getLog().getInfoMessages()).contains(
+                "Searching for all files in '/absolute/path' that match the pattern '**/*.txt'",
+                "-> found 4 files",
+                "Successfully processed file '/one.txt'",
+                "Successfully processed file '/two.txt'",
+                "Skipping file 'not-readable.txt' because it's empty");
+        assertThat(actualResultTwo.hasErrors()).isTrue();
+        assertThat(actualResultTwo.getLog().getErrorMessages()).containsExactly("Errors during parsing",
+                "Skipping file 'empty.txt' because Jenkins has no permission to read the file");
```

Comment:
Move that part into a separate test case

Reply:
Resolved and fixed another minor issue in tests. thanks for the comment